lang/src/org/partiql/lang/eval/NaturalExprValueComparators.kt

```diff
@@ -157,6 +170,13 @@ enum class NaturalExprValueComparators(private val nullOrder: NullOrder) : Compa
         }
     }
 
+    private fun <T> orderLeftRight(order: Order, left: T, right: T): Pair<T, T> {
```

Comment:
Swapping the left and right side ordering is what we want when comparing between two of the **same** data types for `DESC`. But since we also want values of different types to be sorted in descending order, this approach won't work. To resolve this without repeating too much of the logic, we could break the old version of `compare` into its own function and return the inverse result.

See this gist with one approach to reverse the result of comparison for `DESC`: https://gist.github.com/alancai98/c69fc04f8c4c41177c67c8ffa064012c

Reply:
I have committed the code from your gist and changed the tests a bit. Now we are comparing different data types with the following order (Nulls/Missing can be changed by ordering and nulls spec)
* bools, numbers, timestamp, text ... for ascending order
* bag, tuple, array, lob for descending order